cmd/maya-apiserver/cstor-operator/cspc/operations.go

```diff
@@ -0,0 +1,189 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cspc
+
+import (
+	"github.com/golang/glog"
+	nodeselect "github.com/openebs/maya/pkg/algorithm/nodeselect/v1alpha2"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	apiscsp "github.com/openebs/maya/pkg/cstor/newpool/v1alpha3"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (pc *PoolConfig) handleOperations() {
+	// TODO: Disable pool-mgmt reconciliation
+	// before carrying out ant day 2 ops.
+	pc.expandPool()
+	pc.replaceBlockDevice()
+	// Once all operations are executed enable pool-mgmt
+	// reconciliation
+}
+
+// replaceBlockDevice replaces block devices in cStor pools as specified in CSPC.
+func (pc *PoolConfig) replaceBlockDevice() {
+	glog.V(2).Info("block device replacement is not supported yet")
+}
+
+// expandPool expands the required cStor pools as specified in CSPC
+func (pc *PoolConfig) expandPool() error {
+	for _, pool := range pc.AlgorithmConfig.CSPC.Spec.Pools {
+		pool := pool
+		var cspObj *apis.NewTestCStorPool
+		nodeName, err := nodeselect.GetNodeFromLabelSelector(pool.NodeSelector)
+		if err != nil {
+			return errors.Wrapf(err,
+				"could not get node name for node selector {%v} "+
+					"from cspc %s", pool.NodeSelector, pc.AlgorithmConfig.CSPC.Name)
+		}
+
+		cspObj, err = pc.getCSPWithNodeName(nodeName)
+		if err != nil {
+			return errors.Wrapf(err, "failed to csp with node name %s", nodeName)
+		}
+
+		// Pool expansion for raid group types other than striped
+		if len(pool.RaidGroups) > len(cspObj.Spec.RaidGroups) {
+			cspObj = addGroupToPool(&pool, cspObj)
+		}
+
+		// Pool expansion for striped raid group
+		expandStripedGroup(&pool, cspObj)
+
+		_, err = apiscsp.NewKubeClient().WithNamespace(pc.AlgorithmConfig.Namespace).Update(cspObj)
+		if err != nil {
+			glog.Errorf("could not update csp %s: %s", cspObj.Name, err.Error())
+		}
+	}
+
+	return nil
+}
+
+// addGroupToPool adds a raid group to the csp
+func addGroupToPool(cspcPoolSpec *apis.PoolSpec, csp *apis.NewTestCStorPool) *apis.NewTestCStorPool {
+	for _, cspcRaidGroup := range cspcPoolSpec.RaidGroups {
+		cspcRaidGroup := cspcRaidGroup
+		if !isRaidGroupPresentOnCSP(&cspcRaidGroup, csp) {
+			if cspcRaidGroup.Type == "" {
+				cspcRaidGroup.Type = cspcPoolSpec.PoolConfig.DefaultRaidGroupType
+			}
+			csp.Spec.RaidGroups = append(csp.Spec.RaidGroups, cspcRaidGroup)
+		}
+	}
+	return csp
+}
+
+// expandStripedGroup adds newly added block devices to the striped
+// groups present on CSP
+func expandStripedGroup(cspcPoolSpec *apis.PoolSpec, csp *apis.NewTestCStorPool) {
```

Comment:
this can be called as expandExistingStripedGroup..
and accordingly, conditions as well need to be changes as,
```
if getRaidGroupType(cspcGroup, cspcPoolSpec) != "stripe" {
continue
}
if isRaidGroupPresentOnCSP(&cspcGroup, csp) {
continue
}
```